internal/service/service_test.go

```diff
@@ -0,0 +1,130 @@
+package service_test
+
+import (
+	"fmt"
+	"net/http"
+	"testing"
+	"time"
+
+	"github.com/andreipimenov/golang-training-2021/internal/mock"
+	"github.com/andreipimenov/golang-training-2021/internal/model"
+	"github.com/andreipimenov/golang-training-2021/internal/service"
+	"github.com/go-chi/chi/v5"
+	"github.com/rs/zerolog"
+	"github.com/stretchr/testify/suite"
+)
+
+const (
+	invalidTicker  = "ASDASD"
+	ticker         = "AAPL"
+	date           = "2021-07-26"
+	apiKey         = "asdasd"
+	invalidApiKey  = ""
+	invalidDate    = "2029-07-26"
+	badURL         = "dsadasd%s%s"
+	stockAPIFormat = "https://www.alphavantage.co/query?function=TIME_SERIES_DAILY&outputsize=full&symbol=%s&apikey=%s"
+)
+
+type serviceTestSuite struct {
+	suite.Suite
+	clientMock *mock.HTTPClient
+	repoMock   *mock.Repository
+	service    *service.Service
+	router     *chi.Mux
+}
+
+func (suite *serviceTestSuite) SetupTest() {
+	client := &mock.HTTPClient{}
+	repo := &mock.Repository{}
+	s := service.New(&zerolog.Logger{}, repo, apiKey)
```

Comment:
Here you should change `New` construction function signature and pass one more arg `&mock.HTTPClient{}`

Reply:
Currently you don't mock http client and it makes real calls.